src/components/message/MessageInput.tsx

```diff
@@ -1,108 +1,98 @@
 import { Box } from "@twilio-paste/core";
-import { useState, useLayoutEffect, KeyboardEventHandler } from "react";
+import { ChatComposer } from "@twilio-paste/core/chat-composer";
+import {
+  $getRoot,
+  ClearEditorPlugin,
+  CLEAR_EDITOR_COMMAND,
+  COMMAND_PRIORITY_LOW,
+  EditorState,
+  KEY_ENTER_COMMAND,
+  useLexicalComposerContext,
+} from "@twilio-paste/lexical-library";
+import { useEffect } from "react";
 import MessageFile from "./MessageFile";
 
 interface MessageInputProps {
   message: string;
   onChange: (message: string) => void;
-  onKeyPress: KeyboardEventHandler<HTMLInputElement>;
+  onEnterKeyPress: () => void;
   onFileRemove: (file: string) => void;
   assets: File[];
 }
 
-function useWindowSize() {
-  const [size, setSize] = useState(0);
-  useLayoutEffect(() => {
-    function updateSize() {
-      setSize(window.innerWidth);
-    }
-    window.addEventListener("resize", updateSize);
-    updateSize();
-    return () => window.removeEventListener("resize", updateSize);
-  }, []);
-  return size;
+interface EnterKeyPluginProps {
+  onEnterKeyPress: () => void;
 }
 
-function getTextWidth(text: string) {
-  const canvas = document.createElement("canvas");
-  const context = canvas.getContext("2d");
-  if (context !== null && context !== undefined) {
-    context.font = "14px Inter";
-    return context.measureText(text).width;
-  }
-  return 0;
+const EnterKeyPlugin = (props: EnterKeyPluginProps) => {
+  const { onEnterKeyPress } = props;
+  const [editor] = useLexicalComposerContext();
+
+  useEffect(() => {
+    return editor.registerCommand(
+      KEY_ENTER_COMMAND,
+      () => {
+        onEnterKeyPress();
+        return true;
+      },
+      COMMAND_PRIORITY_LOW
+    );
+  }, [editor, onEnterKeyPress]);
+
+  return null;
+};
+
+// when message gets cleared and given it's a prop passed in to MessageInput
+// we need to clear the Lexical editor.
+// TODO: there has to be a simpler way of doing a basic binding like this with Lexical
+
+interface MessagePropPluginProps {
+  message: string;
 }
 
+const MessagePropPlugin = (props: MessagePropPluginProps) => {
+  const { message } = props;
+  const [editor] = useLexicalComposerContext();
+
+  useEffect(() => {
+    if (message === undefined || message === null || message.length === 0) {
```

Comment:
since `message` is a string, this is equivalent to `if(!message)` but then if message becomes a different type in the future maybe the simpler check is less robust 🤷 ?